### DISCUSSION TITLE
Support including debian packages which call `adduser` normal user

### DISCUSSION THREAD

**Discussion by ragazenta** [systemd/mkosi#2235] [Q&A] [answered]
Hi, we have internal debian package which call `adduser` in its `preinst` script. Currently if I include that package in `mkosi.conf` it will fail to build.

```shell
#!/bin/bash -e

adduser --disabled-password --shell /bin/bash --comment "$DESCRIPTION" $USERNAME
```

```shell
Copying files from `/etc/skel' ...
Password: chfn: PAM: Authentication failure
adduser: `/bin/chfn -f XXX YYY xxx' returned error code 1. Exiting.
```

I think somethings wrong with `chfn`.

Fortunately, `useradd` works fine.

**Comment by behrmann** [comment]
If the package is internal, I assume you control it. If that is the case I would recommend moving to a sysuser snippet using `dh_sysuser`.

**Reply by ragazenta** [reply]
Hmm, `dh_sysuser` use `useradd`. It will work fine.

But if I want the user to have "description" or "full name", I need to use `chfn` which currently fail.
My current workaround is to install it in `postscript.chroot` via `dpkg -i`. It works but rather troublesome.

Just curious why `chfn` failed in my case.

**Comment by ragazenta** [comment]
It's really weird, I tried include `mariadb-server` which also call `adduser` in its `preinst` script.  It works as expected, nothing wrong.

**Reply by ragazenta** [reply, marked as answer]
Ok, there is a problem in our package. It should has:
```
Pre-Depends: adduser
```

but it didn't.

Previously we use official debian image and chroot into it, `adduser` is already installed, so we didn't experience this problem before.
